plugins/by-name/lspkind/default.nix

```
@@ -2,93 +2,61 @@
+  name = "lspkind";
+  packPathName = "lspkind.nvim";
+  package = "lspkind-nvim";
+  maintainers = [ lib.maintainers.saygo-png ];
+  description = "VS Code-like pictograms for Neovim LSP completion items.";
 
-    mode = helpers.defaultNullOpts.mkEnum [
-      "text"
-      "text_symbol"
-      "symbol_text"
-      "symbol"
-    ] "symbol_text" "Defines how annotations are shown";
+  # The plugin does not call setup when integrating with cmp, so it has to be conditional.
+  callSetup = false;
 
-    preset = helpers.defaultNullOpts.mkEnum [
-      "default"
-      "codicons"
-    ] "codicons" "Default symbol map";
-
-    symbolMap = helpers.mkNullOrOption (types.attrsOf types.str) "Override preset symbols";
+  # TODO: introduced 2025-07-16: remove after 25.11
+  inherit (import ./deprecations.nix lib) deprecateExtraOptions optionsRenamedToSettings;
 
+  extraOptions = {
     cmp = {
-      enable = mkOption {
+      enable = lib.mkOption {
         type = types.bool;
         default = true;
         description = "Integrate with nvim-cmp";
       };
 
-      maxWidth = helpers.mkNullOrOption types.int "Maximum number of characters to show in the popup";
-
-      ellipsisChar = helpers.mkNullOrOption types.str "Character to show when the popup exceeds maxwidth";
-
-      menu = helpers.mkNullOrOption (types.attrsOf types.str) "Show source names in the popup";
-
       after = helpers.mkNullOrOption types.str "Function to run after calculating the formatting. function(entry, vim_item, kind)";
     };
   };
 
-  config =
-    let
-      doCmp = cfg.cmp.enable && config.plugins.cmp.enable;
-      options = {
-        inherit (cfg) mode preset;
-        symbol_map = cfg.symbolMap;
-      }
-      // (
-        if doCmp then
-          {
-            maxwidth = cfg.cmp.maxWidth;
-            ellipsis_char = cfg.cmp.ellipsisChar;
-            inherit (cfg.cmp) menu;
-          }
-        else
-          { }
-      )
-      // cfg.extraOptions;
-    in
-    mkIf cfg.enable {
-      extraPlugins = [ cfg.package ];
-
-      extraConfigLua = optionalString (!doCmp) ''
-        require('lspkind').init(${lib.nixvim.toLuaObject options})
+  extraConfig = cfg: {
+    assertions = lib.nixvim.mkAssertions "plugins.lspkind" {
+      assertion = cfg.cmp.enable -> config.plugins.cmp.enable;
+      message = ''
+        Cmp integration (cmp.enable) is enabled but the cmp plugin is not.
       '';
-
-      plugins.cmp.settings.formatting.format =
-        if cfg.cmp.after != null then
-          ''
-            function(entry, vim_item)
-              local kind = require('lspkind').cmp_format(${lib.nixvim.toLuaObject options})(entry, vim_item)
-
-              return (${cfg.cmp.after})(entry, vim_item, kind)
-            end
-          ''
-        else
-          ''
-            require('lspkind').cmp_format(${lib.nixvim.toLuaObject options})
-          '';
     };
+
+    plugins.lspkind.luaConfig.content = lib.optionalString (!cfg.cmp.enable) ''
```

I should've spotted this last review, but ideally this'd also be `mkIf`

```suggestion
    plugins.lspkind.luaConfig.content = lib.mkIf (!cfg.cmp.enable) ''
```

Less important here, though.